server/src/main/java/org/elasticsearch/rest/action/document/RestBulkAction.java

```diff
@@ -103,4 +103,9 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
     public boolean supportsContentStream() {
         return true;
     }
+
+    @Override
+    public boolean allowsUnsafeBuffers() {
+        return true;
```

Comment:
We could optimize this and other spots where we return `true` even further by releasing the request at the earliest possible point instead of when sending out the response. I'd push that to a follow up though, as it's not entirely trivial to get this right.